FreeRTOS/FreeRTOS-Plus-TCP
# Updates for FreeRTOS_IP.c /multi

Description
-----------
Added the downward compatibility mode `ipconfigCOMPATIBLE_WITH_SINGLE` which has a global start-up function:
~~~c
BaseType_t FreeRTOS_IPInit( const uint8_t ucIPAddress[ ipIP_ADDRESS_LENGTH_BYTES ],
                            const uint8_t ucNetMask[ ipIP_ADDRESS_LENGTH_BYTES ],
                            const uint8_t ucGatewayAddress[ ipIP_ADDRESS_LENGTH_BYTES ],
                            const uint8_t ucDNSServerAddress[ ipIP_ADDRESS_LENGTH_BYTES ],
                            const uint8_t ucMACAddressP[ ipMAC_ADDRESS_LENGTH_BYTES ] )
~~~

In this mode, only one interface can be used which will have one end-point. No code changes are needed, except adding the module `FreeRTOS_Routing.c`.

Added functions to resolve multicast addresses, both for IPv4 and IPv6.


Test Steps
-----------
Take any existing IPv4 /single application. In FreeRTOSIPConfig.h define:
~~~c
#define ipconfigCOMPATIBLE_WITH_SINGLE   1
~~~
Add FreeRTOS_Routing.c to the project sources, and it should run.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
